hoomd/write/gsd.py

```diff
@@ -177,30 +180,37 @@ def _attach(self):
         super()._attach()
 
     @staticmethod
-    def write(state, filename, filter=All(), log=None):
+    def write(state, filename, filter=All(), mode='wb', log=None):
         """Write the given simulation state out to a GSD file.
 
         Args:
             state (State): Simulation state.
             filename (str): File name to write.
-            filter_ (``hoomd.ParticleFilter``): Select the particles to write.
-            log (``hoomd.logger.Logger``): A ``Logger`` object for GSD logging.
+            filter (`hoomd.ParticleFilter`): Select the particles to write.
+            mode (str): The file open mode. Defaults to `wb`.
+            log (`hoomd.logger.Logger`): Provide log quantities to write.
 
-        Note:
-            The file is always overwritten.
+        The valid file modes for `write` are ``'wb'`` and ``'xb'``.
         """
+        if mode != 'wb' and mode != 'xb':
+            raise ValueError(f"Invalid GSD.write file mode: {mode}")
+
         writer = _hoomd.GSDDumpWriter(state._cpp_sys_def,
                                       filename,
                                       state._get_group(filter),
-                                      True,
+                                      mode,
                                       False)
 
         if log is not None:
-            writer.log_writer = GSDLogWriter(log)
+            writer.log_writer = _GSDLogWriter(log)
         writer.analyze(state._simulation.timestep)
 
     @property
     def log(self):
+        """hoomd.logging.Logger: Provide log quantities to write.
+
+        May be `None`.
+        """
         return self._log
 
     @log.setter
```

Comment:
We should probably fix this to be settable to `None`.

Reply:
Done.